sessionctx/variable/varsutil.go

```diff
@@ -152,7 +152,11 @@ func checkReadOnly(vars *SessionVars, normalizedValue string, originalValue stri
 func checkIsolationLevel(vars *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
 	if normalizedValue == "SERIALIZABLE" || normalizedValue == "READ-UNCOMMITTED" {
 		returnErr := ErrUnsupportedIsolationLevel.GenWithStackByArgs(normalizedValue)
-		if !TiDBOptOn(vars.systems[TiDBSkipIsolationLevelCheck]) {
+		value, err := GetSessionOrGlobalSystemVar(vars, TiDBSkipIsolationLevelCheck)
+		if err != nil {
+			return normalizedValue, err
+		}
+		if !TiDBOptOn(value) {
```

Comment:
Shouldn't this match the corresponding scope of what you are trying to change? i.e. to change the global isolation level, it checks the global scoped skip setting.

This is the behavior of `checkReadOnly`:

https://github.com/pingcap/tidb/blob/dfd358a9882b93195305ba2e9f31e0c89a058290/sessionctx/variable/varsutil.go#L131-L150